feat(mcp): per-server startup timeout (#3182)
openai/codex: docs/config.md

**Note:** Codex may cache the list of tools and resources from an MCP server so that Codex can include this information in context at startup without spawning all the servers. This is designed to save resources by loading MCP servers lazily.

> Each server may set `startup_timeout_ms` to adjust how long Codex waits for it to start and respond to a tools listing. The default is `10000` (10 seconds).

Reviewer: ```suggestion
Each server may set `startup_timeout_ms` to adjust how long Codex waits for it to start and respond to a tools listing. The default is `10_000` (10 seconds).
```

command = "npx"
args = ["-y", "mcp-server"]
env = { "API_KEY" = "value" }
# Optional: override the default 10s startup timeout

> startup_timeout_ms = 20000

Reviewer: Note TOML supports `_` to break up int values.

```suggestion
startup_timeout_ms = 20_000
```

| `mcp_servers.<id>.command` | string | MCP server launcher command. |
| `mcp_servers.<id>.args` | array<string> | MCP server args. |
| `mcp_servers.<id>.env` | map<string,string> | MCP server env vars. |

> | `mcp_servers.<id>.startup_timeout_ms` | number | Startup timeout in milliseconds (default: 10000). Timeout is applied both for initializing MCP server and initially listing tools. |

Reviewer: ```suggestion
| `mcp_servers.<id>.startup_timeout_ms` | number | Startup timeout in milliseconds (default: 10_000). Timeout is applied both for initializing MCP server and initially listing tools. |
```
